File: CS/DocumentVariablesExample/Weather.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using System.Net;
using System.Web;
using DevExpress.Utils;
using System.Windows.Browser;
using DevExpress.Xpf.Collections;

namespace DocumentVariablesExample
{
    class Weather
    {

        public static Conditions GetCurrentConditions(string _city, ArrayList _conditionsdata)
        {
            List<Conditions> weather_conditions = new List<Conditions>();

            var query =
                    from Conditions conds in _conditionsdata
                    where conds.City == _city
                    select conds;

            foreach (Conditions c in query)
                weather_conditions.Add(c);

            return weather_conditions[0];
        }

  
    }
    public class Conditions
    {
        string city = "No Data";
        string dayOfWeek = DateTime.Now.DayOfWeek.ToString();
        string condition = "No Data";

        public string City
        {
            get { return city; }
            set { city = value; }
        }

        public string Condition
        {
            get { return condition; }
            set { condition = value; }
        }

        public Conditions(string city, string condition) {
            this.City = city;
            this.Condition = condition;
        }
    }
}

File: VB/DocumentVariablesExample/Weather.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Xml
Imports System.Net
Imports System.Web
Imports DevExpress.Utils
Imports System.Windows.Browser
Imports DevExpress.Xpf.Collections

Namespace DocumentVariablesExample
	Friend Class Weather

		Public Shared Function GetCurrentConditions(ByVal _city As String, ByVal _conditionsdata As ArrayList) As Conditions
			Dim weather_conditions As New List(Of Conditions)()

			Dim query = _
				From conds As Conditions In _conditionsdata _
				Where conds.City = _city _
				Select conds

			For Each c As Conditions In query
				weather_conditions.Add(c)
			Next c

			Return weather_conditions(0)
		End Function


	End Class
	Public Class Conditions
		Private city_Renamed As String = "No Data"
		Private dayOfWeek As String = DateTime.Now.DayOfWeek.ToString()
		Private condition_Renamed As String = "No Data"

		Public Property City() As String
			Get
				Return city_Renamed
			End Get
			Set(ByVal value As String)
				city_Renamed = value
			End Set
		End Property

		Public Property Condition() As String
			Get
				Return condition_Renamed
			End Get
			Set(ByVal value As String)
				condition_Renamed = value
			End Set
		End Property

		Public Sub New(ByVal city As String, ByVal condition As String)
			Me.City = city
			Me.Condition = condition
		End Sub
	End Class
End Namespace
